m4/configure_boost.m4

```diff
@@ -14,7 +14,7 @@ AC_ARG_VAR(BOOST_ROOT,[the location of the Boost installation that is to be used
 #$as_unset boost_cv_version
 #BOOST_REQUIRE([1.57.0],[AC_MSG_WARN([could not find system Boost library, using bundled Boost library])])
 if test x"$USING_BUNDLED_BOOST" = xyes ; then
-  if test "$LIBMESH_ENABLED" = yes ; then
+  if test "$LIBMESH_HAS_BOOST" = yes ; then
     AC_MSG_ERROR([IBAMR is configured to use a bundled Boost library, but when IBAMR is also configured to use libMesh, IBAMR and libMesh both must use the same (external) Boost library])
```

Comment:
I don't think this error message is right: we support using libMesh's bundled copy of boost, no?

Reply:
At least at some point we used stuff from boost not provided by libMesh's bundled copy. I do not know if that is currently the case.

Reply:
I think this PR is ready once this error message is clarified.